Classes/TextUtils.cs

```diff
@@ -24,19 +24,106 @@ namespace RadioDld
+        private static Regex removeDaySuffix = new Regex(reDaySuffix, RegexOptions.IgnoreCase);
+
+        // Set day range for Similar date check
+        private static int iSimilarDateDayRange = 6;
+        //// private static int iSimilarDateDayRange = 3;
+
+        /* OLD FUNCTION for record
+        *  public static string oldStripDateFromName(string name, DateTime stripDate)
+        *  {
+        *      // Use regex to remove a number of different date formats from episode titles.
+        *      // Will only remove dates with the same month & year as the programme itself, but any day of the month
+        *      // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
+        *      Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+        *      if (matchStripDate.IsMatch(name))
+        *      {
+        *          name = matchStripDate.Match(name).Groups["name"].ToString();
+        *      }
+        *      return name;
+        *  }
+        */
+
         public static string StripDateFromName(string name, DateTime stripDate)
         {
             // Use regex to remove a number of different date formats from episode titles.
-            // Will only remove dates with the same month & year as the programme itself, but any day of the month
-            // as there is sometimes a mismatch of a day or two between the date in a title and the publish date.
-            Regex matchStripDate = new Regex("\\A(" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/\\d{2} ?-? )?(?<name>.*?)( ?:? (\\d{2}/" + stripDate.ToString("MM", CultureInfo.InvariantCulture) + "/" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "|((Mon|Tue|Wed|Thu|Fri) )?(\\d{1,2}(st|nd|rd|th)? )?(" + stripDate.ToString("MMMM", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("MMM", CultureInfo.InvariantCulture) + ")( \\d{1,2}(st|nd|rd|th)?| (" + stripDate.ToString("yy", CultureInfo.InvariantCulture) + "|" + stripDate.ToString("yyyy", CultureInfo.InvariantCulture) + "))?))?\\Z");
+            // Will only remove dates within a set range eitherside of the programme date.
```

Comment:
My fault but we may as well fix it while we are here - this should actually be a `///` xml comment before the function.